WordPress/gutenberg
# Experiment to connect HTML attributes with custom fields

## What?
This is an experiment to explore the possibility of connecting HTML attributes with any source like meta/custom fields.

I made a video explaining it a bit deeper:

https://www.loom.com/share/ce428827a81248a297a1b9fa0b091dbc?sid=f9665430-b6c3-41f7-b873-9f852d1e619d

## How?
To make it work, I incorporated the changes from [this pull request](https://github.com/WordPress/wordpress-develop/pull/4956) in `wordpress/develop` because I wanted to use the `set_inner_markup` function. This will eventually be included in core so it is not relevant for this pull request.

The important part is the logic included in the `blocks.php`, that is reusing the code that connects block attributes and custom fields.

## Testing Instructions

1. Add a `data-wp-dynamic-content` attribute to any HTML tag with an object similar to:
```json
{
  "html_attributes": {
    "inner_content": {
      "source": "meta_fields",
      "key": "custom_text"
    },
    "href": {
      "source": "meta_fields",
      "key": "custom_link"
    }
  }
}
```

2. Register the `custom_text` and `custom_link` custom fields using `register_post_meta` for example.
3. Check that the HTML attributes or inner content is replaced in the frontend by the value of the custom fields.
